### PR TITLE
Removes size validation in TryGetDirectoryOffset, adds tests

### LINKED ISSUE #1664: System.Reflection.Metadata fails to read offset of DataDirectories spanning across multiple sections
If an assembly contains any DataDirectory which is in more than one section we throw "System.BadImageFormatException : Section too small.". While I agree we should not have any assemblies like that we unfortunately already shipped one which has this issue.

I believe we can still calculate the offset without checking ranges (or we can check the global range only)

Example can be found here:
https://github.com/krwq/corefx/tree/pereaderdatadir

This is basically:
```csharp
using (FileStream fs = new FileStream(@"C:\Windows\Microsoft.NET\Framework\v4.0.30319\Microsoft.JScript.dll", FileMode.Open, FileAccess.Read, FileShare.ReadWrite | FileShare.Delete))
{
    using (PEReader reader = new PEReader(fs, PEStreamOptions.LeaveOpen | PEStreamOptions.PrefetchEntireImage | PEStreamOptions.PrefetchMetadata))
    {
        int offset;
        if (reader.PEHeaders.TryGetDirectoryOffset(reader.PEHeaders.PEHeader.CertificateTableDirectory, out offset))
        {

        }
    }
}
```

### PR DESCRIPTION
**Pull request by tmat**
Fixes #1664

### CODE REVIEW

**Comment by tmat** [comment]
@nguerrera

**Comment by tmat** [comment]
FYI @krwq

File: src/System.Reflection.Metadata/tests/Resources/TestResources.cs [hunk 1]
```diff
@@ -61,6 +62,101 @@ internal static class PortablePdbs
...
+            {
+                return _dirBuilder;
+            }
+        }
+
+        private static ImmutableArray<byte> GenerateImage()
+        {
+            var peBuilder = new TestPEBuilder();
+
+            BlobBuilder peImageBuilder = new BlobBuilder();
```
Comment (nguerrera) [inline src/System.Reflection.Metadata/tests/Resources/TestResources.cs]: nit: use `var`

**Comment by nguerrera** [comment]
:+1:

**Comment by krwq** [comment]
Thanks @tmat! LGTM

**Pull request merged**